gax-java/gax-grpc/src/main/java/com/google/api/gax/grpc/InstantiatingGrpcChannelProvider.java

```diff
@@ -410,6 +473,97 @@ ChannelCredentials createMtlsChannelCredentials() throws IOException, GeneralSec
     return null;
   }
 
+  @VisibleForTesting
+  boolean isGoogleS2AEnabled() {
+    String S2AEnv = envProvider.getenv(S2A_ENV_ENABLE_USE_S2A);
+    return Boolean.parseBoolean(S2AEnv);
+  }
+
+  @VisibleForTesting
+  boolean shouldUseS2A() {
+    // If EXPERIMENTAL_GOOGLE_API_USE_S2A is not set to true, skip S2A.
+    if (!isGoogleS2AEnabled()) {
+      return false;
+    }
+
+    // If {@code mtlsEndpoint} is not set, skip S2A.
+    if (mtlsEndpoint.isEmpty()) {
+      return false;
+    }
```

Comment:
I want to double check this behavior with you. 

By default for GAPICs, we do set a default mTLS value for both [gRPC](https://github.com/googleapis/google-cloud-java/blob/816c6150f44110635e30fa9648a2502cd36a8ee1/java-speech/google-cloud-speech/src/main/java/com/google/cloud/speech/v1/stub/SpeechStubSettings.java#L381) and [HttpJson](https://github.com/googleapis/google-cloud-java/blob/816c6150f44110635e30fa9648a2502cd36a8ee1/java-speech/google-cloud-speech/src/main/java/com/google/cloud/speech/v1/stub/SpeechStubSettings.java#L393).

This is something that is auto populated for all GAPIC clients. It's in the generator code: https://github.com/googleapis/sdk-platform-java/blob/c8e3941ef6f5bd1236f5ceedfd488e5113928471/gapic-generator-java/src/main/java/com/google/api/generator/gapic/composer/common/AbstractServiceStubSettingsClassComposer.java#L1180-L1192